internal/pgadmin/users.go

```diff
@@ -105,7 +137,57 @@ with create_app().app_context():
 
         db.session.add(user)
         db.session.commit()
-`
+
+        # Set the cluster and host name variable.
+        clustername = "%s"
+        hostname = "%s"
+
+        # Get or create the group as necessary
+        group = (
+            ServerGroup.query.filter_by(user_id=user.id).first() or
+            ServerGroup()
+        )
+        group.name = "Crunchy PostgreSQL Operator"
+        group.user_id = user.id
+        db.session.add(group)
+        db.session.commit()
+
+        # Get the existing server connection, if there is one.
+        oldserver = Server.query.filter_by(
+            servergroup_id=group.id,
+            user_id=user.id,
+        ).first()
+
+        # Create a server connection with the current required values.
+        server = Server()
+        server.name = clustername
+        server.servergroup_id = group.id
+        server.user_id = user.id
+        server.ssl_mode = "prefer"
+        server.host = hostname
+        server.username = user.email
+        # Save the encrypted server password.
+        server.password = encrypt(data['password'], data['password'])
+        server.maintenance_db = "postgres"
+        server.port = %s
+
+        # If there is no existing server connection, create the new one.
+        if oldserver is None:
+            db.session.add(server)
+            db.session.commit()
+
+        # Otherwise, only recreate a server connection if the existing
+        # connection does not already match the existing server connection.
+        # Here we need to check the individual values that may change because
+        # the Server __eq__() method is not implemented.
+        elif (
+          oldserver.port != server.port or
```

Comment:
:+1: Implemented, initial testing seems good.

The user can't modify most settings via the UI while the server is connected, with the exception of the server connection name. As for the `ssl_mode`, I think that any changes should be maintained as a separate server connection, which will keep the behavior consistent with naming, etc. 

:seedling: It may be worth considering more granular configuration later, but I think that's out of scope for this story.